src/main/scala/com/datastax/gatling/plugin/model/DseGraphStatements.scala

```diff
@@ -61,60 +68,52 @@ case class GraphFluentStatementFromScalaLambda(lambda: Session => GraphStatement
   *
   * @param sessionKey Place a GraphTraversal in your session with this key name
   */
-case class GraphFluentSessionKey(sessionKey: String) extends DseGraphStatement {
+case class GraphFluentSessionKey(sessionKey: String)
+  extends DseGraphStatement[FluentS, FluentB] {
 
-  def buildFromSession(gatlingSession: Session): Validation[GraphStatement] = {
+  def buildFromSession(gatlingSession: Session): Validation[FluentB] = {
 
     if (!gatlingSession.contains(sessionKey)) {
       throw new DseGraphStatementException(s"Passed sessionKey: {$sessionKey} does not exist in Session.")
     }
 
     Try {
-      DseGraph.statementFromTraversal(gatlingSession(sessionKey).as[GraphTraversal[_, _]])
+      FluentS.builder(gatlingSession(sessionKey).as[GraphTraversal[_, _]])
     } match {
-      case TrySuccess(stmt) => stmt.success
+      case TrySuccess(builder) => builder.success
       case TryFailure(error) => error.getMessage.failure
     }
   }
-
 }
 
 /**
   * Set/Bind Gatling Session key/vals to GraphStatement
   *
-  * @param statement   SimpleGraphStatement
+  * @param builder   SimpleGraphStatementBuilder
   * @param sessionKeys Gatling session param keys mapped to their bind name, to allow name override
   */
-case class GraphBoundStatement(statement: SimpleGraphStatement, sessionKeys: Map[String, String]) extends DseGraphStatement {
+case class GraphBoundStatement(builder: ScriptB, sessionKeys: Map[String, String])
```

Comment:
we could leverage scala alias types (`type A = String, type B = String`) within the context of this class to not operate on Strings (`Map[String, String]`), but have more strong typed API (`Map[A,B]`). Similar for other classes

Reply:
I'm not completely sure I follow.  How is this:

```scala
type A = String
type B = String

case class GraphBoundStatement(builder: ScriptB, sessionKeys: Map[A,B])
  extends DseGraphStatement[ScriptS, ScriptB] {
```

offer more or less type information than this:

```scala
case class GraphBoundStatement(builder: ScriptB, sessionKeys: Map[String, String])
  extends DseGraphStatement[ScriptS, ScriptB] {
```
?

Or maybe I'm misunderstanding your point...

Reply:
The `type A = String` is just an example. This `A` and `B` will become descriptive names of the values that his map is carrying.

Reply:
The existing type (Map[String,String]) actually reflects a legit constraint of the code.  For each entry in the input map we pull an object out of the Gatling session and then add it as a query param on the builder.  We need a String to get an entry out of the session and the setQueryParam() call on the builder expects a String to identify the name:

```scala
      sessionKeys foreach { k =>
        k match {
          case (k, v) => builder.setQueryParam(v, gatlingSession(k).as[Object])
          case _ => throw new RuntimeException(s"Observed ${k} instead of expected key-value pair")
        }
      }
```

I suppose we could relax these constraints and call toString() on the key and value we get out of a map entry before using it in this way.. but I'm not sure we gain much by that increase in complexity.

What do you think?

Reply:
the type alias is not creating a new type, it justs gives this type a different name - you can still pass the `String` there but looking at the method you don't need to guess what this `string` is carrying because that info is in the type alias name.